pkg/web/static/js/main.js

```diff
@@ -65,4 +65,23 @@ const updateDatetimes = () => {
   });
 };
 
-document.body.addEventListener('htmx:afterSettle', updateDatetimes);
\ No newline at end of file
+document.body.addEventListener('htmx:afterSettle', updateDatetimes);
+
+// Disable submit button after form submission and enable it after a request.
+const submitBtn = document.querySelector('.submit-btn');
+const submitBtnText = document.querySelector('.submit-btn-text');
+
+function disableSubmitBtn() {
+  document.body.addEventListener('submit', () => {
+    submitBtnText?.classList.add('hidden');
+    submitBtn?.setAttribute('disabled', 'disabled');
+  });
+};
+
+function enableSubmitBtn() {
+  submitBtnText?.classList.remove('hidden');
+  submitBtn?.removeAttribute('disabled');
+};
+
+document.body.addEventListener('htmx:afterSettle', disableSubmitBtn);
```

Comment:
The buttons appear in the `DOM` after an htmx swap, so this is why we're listening for a different events when `disabling` and `re-enabling` the buttons.